File: Práctica 1/Ruby/test_p1.rb
# encoding: UTF-8

require_relative 'directions'
require_relative 'orientation'
require_relative 'game_character'
require_relative 'weapon'
require_relative 'shield'
require_relative 'dice'
require_relative 'game_state'

module Irrgarten

    class TestP1

        def self.main

            puts Directions::UP

            weapon = Weapon.new(4.5, 7)
            shield = Shield.new(7.5, 5)

            puts weapon.to_s
            puts shield.to_s

            gameState = GameState.new("", "", "", 1, false, "Log")

            puts "Jugador Actual: " + gameState.players
            puts "Laberinto: " + gameState.labyrinth

            for i in 0..99

                puts "Posición Aleatoria: " + Dice.random_pos(10).to_s
                puts "Quién empieza: " + Dice.who_starts(10).to_s
                puts "Inteligencia Aleatoria: " + Dice.random_intelligence.to_s
                puts "Fuerza Aleatoria: " + Dice.random_strength.to_s
                puts "¿Se resucita al jugador? " + Dice.resurrect_player.to_s
                puts "Recompensa de Armas: " + Dice.weapons_reward.to_s
                puts "Recompensa de Escudos: " + Dice.shields_reward.to_s
                puts "Recompensa de Salud: " + Dice.health_reward.to_s
                puts "Potencia de Arma: " + Dice.weapon_power.to_s
                puts "Potencia de Escudo: " + Dice.shield_power.to_s
                puts "Usos Restantes: " + Dice.uses_left.to_s
                puts "Intensidad: " + Dice.intensity(10).to_s
                puts "¿Se descarta el elemento? " + Dice.discard_element(10).to_s

            end

        end

    end

end

Irrgarten::TestP1.main

File: Práctica 1/Java/IrrgartenP1/src/irrgarten/TestP1.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author aloncraftmc
 */
public class TestP1 {
    
    public static void main(String[] args){
        
        System.out.println(Directions.UP);
        
        Weapon weapon = new Weapon(4.5f, 7);
        Shield shield = new Shield(7.5f, 5);
        
        System.out.println(weapon);
        System.out.println(shield);
        
        GameState gameState = new GameState("", "", "", 1, false, "Log");
        
        System.out.println("Jugador Actual: " + gameState.getPlayers());
        System.out.println("Laberinto: " + gameState.getLabyrinth());
        
        for(int i = 0; i < 100; i++){
            
            System.out.println("Posición Aleatoria: " + Dice.randomPos(10));
            System.out.println("Quién Empieza: " + Dice.whoStarts(10));
            System.out.println("Inteligencia Aleatoria: " + Dice.randomIntelligence());
            System.out.println("Fuerza Aleatoria: " + Dice.randomStrength());
            System.out.println("¿Se resucita al jugador? " + Dice.resurrectPlayer());
            System.out.println("Recompensa de Armas: " + Dice.weaponsReward());
            System.out.println("Recompensa de Escudos: " + Dice.shieldsReward());
            System.out.println("Recompensa de Salud: " + Dice.healthReward());
            System.out.println("Potencia de Arma: " + Dice.weaponPower());
            System.out.println("Potencia de Escudo: " + Dice.shieldPower());
            System.out.println("Usos Restantes: " + Dice.usesLeft());
            System.out.println("Intensidad: " + Dice.intensity(10));
            System.out.println("¿Se descarta el elemento? " + Dice.discardElement(10));
            
        }
        
    }
    
}
